sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py

```diff
@@ -280,3 +280,22 @@ def _validate_component(self, deployment: Deployment, orchestrators: OperationOr
                 name, _ = parse_prefixed_name_version(deployment.job_definition.component)
                 deployment.job_definition.name = name
             deployment.job_definition.component_id = component_id
+        elif(deployment.job_definition.job):
+            if isinstance(deployment.job_definition.job, str):
```

Comment:
If this needs to go faster please make sure tests cases are added to the SDK layer , preferably unit tests!

Reply:
this [work item](https://msdata.visualstudio.com/DefaultCollection/Vienna/_workitems/edit/2120336/) has been added as a child of the open task. It will handle sdk coverage test